### DISCUSSION TITLE
请问如何了解.vim 中的set对应.lua中的.opt还是.g

### DISCUSSION THREAD

**Discussion by allworldg** [nshen/learn-neovim-lua#54] [Q&A] [answered]
我看了`:help vim.opt`和`:help vim.g`说明，对于大部分set命令都可以用vim.opt替代。但是我应该如何区分呢？
例如 `:help mouse`
![image](https://user-images.githubusercontent.com/48782874/178378360-a78c767f-2501-4639-9444-90a9504a8c74.png)
这里显示global是否意味着应该用`vim.g.mouse`？(虽然这样用会失效）

**Comment by nshen** [comment]
<img width="849" alt="image" src="https://user-images.githubusercontent.com/181506/178408719-ae36c54d-384e-4bad-9eca-f6353fb10a4d.png">

我只知道文档里有 `:set aaa=bbb` 的应该可以改成 `vim.opt.aaa = bbb`

vim.g 应该很少见，可能是可以自己定义全局变量到里边，我也不太清楚，我觉得各种前缀挺混乱的，不知道有没有人可以解释解释。

**Comment by allworldg** [comment, marked as answer]
我觉得算是总结得还不错的说法
`:help` option-list will take you to a list of options with a short description. `:help` E355 is where you can find the list with the full description of every option.

In lua 95% of the time you'll want to use` :help` vim.opt for setting config values. Do read the help page because `vim.opt `it is pure magic.

Mapleader and plugin variables are under` vim.g.`

Buffer local variables are under `vim.bo`.

And env variables are in vim.env.
https://www.reddit.com/r/neovim/comments/r6ar6y/looking_for_list_of_all_vim_api_options/
